src/plonk_recursion.rs

```diff
@@ -87,20 +91,20 @@ pub fn recursive_verification_circuit<
     let old_proofs = make_old_proofs(&mut builder, num_old_proofs, degree_pow);
 
     // Flatten the list of public input openings.
-    let o_public_inputs: Vec<Target> = proof
-        .o_public_inputs
-        .iter()
-        .cloned()
-        .flat_map(|opening_set| opening_set.o_wires)
-        .collect();
+    let o_public_inputs: Option<Vec<Target>> = proof.o_public_inputs.as_ref().map(|pis| {
+        pis.iter()
+            .cloned()
+            .flat_map(|opening_set| opening_set.o_wires)
+            .collect()
+    });
 
-    verify_assumptions::<C, InnerC>(&mut builder, degree_pow, &public_inputs, &o_public_inputs);
+    // verify_assumptions::<C, InnerC>(&mut builder, degree_pow, &public_inputs, &o_public_inputs);
```

Comment:
Was this failing?

Reply:
Yes I get an `index out of bounds` error in `verify_assumptions`. I uncommented it and will try to fix it.